### PR TITLE
Lift to SpeziLLM

### PR DESCRIPTION
**Pull request by philippzagar**
# Lift to SpeziLLM

## :recycle: Current situation & Problem
Dependent on old version of SpeziLLM

## :gear: Release Notes 
- Update to SpeziLLM version 0.7.0

## :books: Documentation
Proper docs included

## :white_check_mark: Testing
All tests passing

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: LLMonFHIR/FHIR Display/InspectResourceView.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Display/InspectResourceView.swift:123]: Could we even use the unknown error type from SpeziLLM here and in other places?

File: LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift:32]: Would it be possible to sue [`@Bindable`](https://developer.apple.com/documentation/swiftui/bindable) here to create the binding of the Observable type?

File: LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift:47]: Same as in the documentation in Spezi LLM: The risk of showing an empty Chat view here that the user could interact with is not really ideal. Maybe there is a way that this could be solved in a reusable manner?

File: LLMonFHIR/FHIR Interpretation/FHIRInterpretationFunction.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Interpretation/FHIRInterpretationFunction.swift:38]: Nice to see the parallelization here!

File: LLMonFHIR/FHIR Interpretation/FHIRInterpretationModule.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Interpretation/FHIRInterpretationModule.swift]: As far as I can see this, a change in the model type at runtime won't update this in the functions. It doesn't seem possible to move the update logic somewhere further down in the logic of the functions. Would probably require a listener on the changes in the user defaults and then regenerate the function? But even then it might not be picked up by the related modules?

File: LLMonFHIR/FHIR Interpretation/FHIRMultipleResourceInterpreter.swift
Comment (PSchmiedmayer) [inline LLMonFHIR/FHIR Interpretation/FHIRMultipleResourceInterpreter.swift:1]: Greatly simplifies the code here; great to see!

File: LLMonFHIR.xcodeproj/project.pbxproj
Comment (PSchmiedmayer) [inline LLMonFHIR.xcodeproj/project.pbxproj]: TODO to update them to the release versions once they have been merged and tagged.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for all the updates! I took a look at the changes and added some comments here and there about suggestion improvements. 

Excited to see this moving up the hierarchy of projects and to see the LLM advancements showing up in LLM on FHIR; great job @philippzagar! 🚀

File: LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift
Comment (philippzagar) [reply LLMonFHIR/FHIR Display/MultipleResourcesChatView.swift]: Sadly not, I couldn't get that to work in my tries. The optionality of the `llm` is cruicial

**Comment by philippzagar** [comment]
@PSchmiedmayer This PR can be merged after https://github.com/StanfordSpezi/SpeziFHIR/pull/17 is merged and the dependency within LLMonFHIR is updated to the tagged release 🚀

**Pull request merged**
